### DISCUSSION TITLE
How to get coordinates stored as geoip from Firebase

### DISCUSSION THREAD

**Discussion by Qoyyuum** [Budibase/budibase#6686] [Help] [answered]
Hi all,

I'm working on transforming the geoip from Firebase to Budibase but I'm not sure how to. After running a simple query to Firebase, I get a `[object Object]` for the coordinates as its set as a `geoip` type.

```
{
  "coordinates": "[object Object]",
  "picture": "https://randomuser.me/api/portraits/women/66.jpg",
  "address": "Somewhere in the world",
  "name": "Susan Smith"
}
```

I tried `return [data.coordinates[0], data.coordinates[1]]` and that didn't work either. Is there a way to have the latitude and longitude fields in the map component to recognize that the field has both and should select the first for latitude and the second for the longitude?

Please advise.

Edit: Just to share after running the query in Budibase, it sees and can't determine the type for the field "coordinates" and couldn't find a proper data type for coordinates.

![image](https://user-images.githubusercontent.com/971309/178769145-fbc6773b-a1ca-41af-9522-f7f81295f998.png)

**Comment by melohagan** [comment]
Hi @Qoyyuum 

To confirm, is coordinates of type **GeoPoint** ? If not it would help to have a link to the relevant Firebase doc for geoip types - I couldn't find it myself after a quick search. 

Either way it may be possible to do something like `data.coordinates['latitude']`

Otherwise we'll need to make a code change within the Firestore connector as outlined here: https://stackoverflow.com/questions/53799346/how-to-convert-geopoint-in-firestore-to-latlng

**Reply by Qoyyuum** [reply]
Yes, the coordinates are of type GeoPoint. I can put latitude and longitude in the GeoPoint type:

![image](https://user-images.githubusercontent.com/971309/179519936-b87ecc0c-ef2f-420e-a986-c78665761931.png)

But unable to access any of them. I have tried

```
data.coordinates
data['coordinates']
data.coordinates[0]
data['coordinates'][0]
```

None of them worked.

**Reply by Qoyyuum** [reply]
It's also possible that the GeoPoint data type is not recognized as a valid data type from Budibase's perspective. Maybe set it as a int array or something?

![image](https://user-images.githubusercontent.com/971309/179520331-31676f45-5dd7-47aa-aa38-e5353cecc82b.png)

**Comment by melohagan** [comment, marked as answer]
Hi @Qoyyuum 

I've had a closer look and realised no code change is needed. You just need to use the transformer to map the co-ordinates like so:

```
return data.map(item => ({
	...item,
	latitude: item.geoip._latitude,
	longitude: item.geoip._longitude
}))
```

**data** represents all of the rows, which is why `data.coordinates` doesn't work. In your case replace where I have *geoip* with *coordinates*.

![Screenshot 2022-07-18 at 15 01 16](https://user-images.githubusercontent.com/101575380/179528437-d0345a21-d89d-4b92-b729-588f607a30e1.png)

Then if you don't want to include the coordinates/geoip field in your schema then you can click the `X` within the Schema tab:
![Screenshot 2022-07-18 at 15 02 06](https://user-images.githubusercontent.com/101575380/179528603-38216f86-f9b3-4dfb-a09b-e4b69a110780.png)

Save your query.

**Reply by Qoyyuum** [reply]
Thank you so much! Wow that was tricky!
